### PR TITLE
feat(visionos): Vision Pro support

### PR DESCRIPTION
**Pull request by NathanWalker**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/tools/notes/CONTRIBUTING.md#commit-messages.
- [ ] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [x] All existing tests are passing: https://github.com/NativeScript/NativeScript/blob/master/tools/notes/DevelopmentWorkflow.md#running-unit-tests-application.
- [x] Tests for the changes are included - https://github.com/NativeScript/NativeScript/blob/master/tools/notes/WritingUnitTests.md.

## What is the current behavior?

Core did not run directly on visionos.

## What is the new behavior?

Core can now be used on visionos.

### CODE REVIEW

File: migrations.json
Comment (rigor789) [inline migrations.json:1]: Not sure we need to keep these, can't remember...

File: packages/core/config/config.interface.ts [hunk 1]
```diff
@@ -20,11 +20,20 @@ interface IConfigIOS extends IConfigPlatform {
...
+		/**
+		 * Remote package url
+		 */
+		repositoryURL?: string;
+		version?: string;
+		/**
+		 * Local package path relative to project directory
+		 */
+		path?: string;
 	}>;
```
Comment (rigor789) [inline packages/core/config/config.interface.ts:32]: nit: Might be better to copy the same exact types from the trapeze definitions:

```ts
export interface IosRemoteSPMPackageDefinition {
  name: string;
  libs: string[];
  repositoryURL: string;
  version: string;
}

export interface IosLocalSPMPackageDefinition {
  name: string;
  libs: string[];
  path: string;
}

export type IosSPMPackageDefinition = IosRemoteSPMPackageDefinition | IosLocalSPMPackageDefinition;
```

And change this type to `Array<IosSPMPackageDefinition>`. The main difference between the two approaches is that we'd get a warning/error when trying to use `version` or `repositoryURL` in a local SPM package definition (and the other way around, trying to use `path` in a remote definition).

File: packages/core/platforms/ios/TNSWidgets.xcframework/ios-arm64/dSYMs/TNSWidgets.framework.dSYM/Contents/Resources/Relocations/aarch64/TNSWidgets.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
+---
+triple:          'arm64-apple-darwin'
+binary-path:     '/Users/nstudio/Documents/github/NativeScript/NativeScript/packages/ui-mobile-base/ios/TNSWidgets/build/Release-iphoneos/TNSWidgets.framework/TNSWidgets'
```
Comment (rigor789) [inline packages/core/platforms/ios/TNSWidgets.xcframework/ios-arm64/dSYMs/TNSWidgets.framework.dSYM/Contents/Resources/Relocations/aarch64/TNSWidgets.yml:3]: This is auto-generated, but I do wonder why it's here now (and if having that path is an issue for consuming these dSYMs).

**Review by rigor789 — APPROVED** [review]
LGTM, only added some notes/nits.

File: migrations.json
Comment (NathanWalker) [reply migrations.json:1]: I'll remove it as not needed.

File: packages/core/config/config.interface.ts (hunk 1 above)
Comment (NathanWalker) [reply packages/core/config/config.interface.ts]: Thank you, modified to match that now.

File: packages/core/platforms/ios/TNSWidgets.xcframework/ios-arm64/dSYMs/TNSWidgets.framework.dSYM/Contents/Resources/Relocations/aarch64/TNSWidgets.yml (hunk 2 above)
Comment (NathanWalker) [reply packages/core/platforms/ios/TNSWidgets.xcframework/ios-arm64/dSYMs/TNSWidgets.framework.dSYM/Contents/Resources/Relocations/aarch64/TNSWidgets.yml:3]: Good question, I'm going to remove it from repo and when regenerated prior to release we can check.

**Pull request merged**